api/src/main/java/org/openmrs/module/facilityregistry/utils/FacilityRegistryUtils.java

```diff
@@ -0,0 +1,137 @@
+package org.openmrs.module.facilityregistry.utils;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.instance.model.api.IAnyResource;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.Location;
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.ResourceType;
+import org.hl7.fhir.r4.model.Type;
+import org.openmrs.module.facilityregistry.FacilityRegistryConstants;
+import org.openmrs.module.facilityregistry.api.FhirOrganizationService;
+import org.openmrs.module.facilityregistry.task.FacilityRegistryTask;
+import org.openmrs.module.fhir2.api.FhirLocationService;
+import org.openmrs.module.fhir2.api.FhirService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class FacilityRegistryUtils {
+	
+	@Autowired
+	private FhirLocationService locationService;
+	
+	@Autowired
+	private FhirOrganizationService organizationService;
+	
+	private static final Logger log = LoggerFactory.getLogger(FacilityRegistryUtils.class);
+	
+	/**
+	 * Saves/Updates Fhir Location into the OpenMRS database
+	 * 
+	 * @param searchBundle Bundle fetched from the Facility Registry Server
+	 */
+	public Resource saveFhirLocation(Bundle searchBundle, IGenericClient fhirClient) {
+		for (Bundle.BundleEntryComponent entry : searchBundle.getEntry()) {
+			if (entry.hasResource()) {
+				if (ResourceType.Location.equals(entry.getResource().getResourceType())) {
+					Location newLocation = (Location) entry.getResource();
+					// tag the Location from Facility Registry as mCSD Location
+					newLocation.getMeta().addTag(FacilityRegistryConstants.FACILITY_REGISTRY_LOCATION_FHIR_SYSTEM,
+					    FacilityRegistryConstants.FACILITY_REGISTRY_LOCATION,
+					    FacilityRegistryConstants.FACILITY_REGISTRY_LOCATION);
+					// tag Laboratory locations as mCSD Laboratory
+					if (newLocation.hasType("Laboratory")) {
+						newLocation.getMeta().addTag("mCSD Laboratory", "mCSD Laboratory", "mCSD Laboratory");
+					}
+					return saveOrUpdateLocation(newLocation);
```

Comment:
and also we cant use a return with in the for loop , it will terminate the loop and dont save the rest of the resources with in openmrs